### PR TITLE
Add RTL style overrides, ensure RTL CSS is loaded on RTL sites

### LINKED ISSUE #2483: RTL Audit
- [x] Change to a right to left language and browse the site
- [x] Log issues as comments below
- [x] Fix issues

### PR DESCRIPTION
**Pull request by ryelle**
Fixes #2483. This updates the style.css registration to create an RTL replacement on RTL sites — this means the `style-index-rtl.css` file will be used instead. This PR also includes a few overrides specific to RTL sites, in the `_rtl.scss` file.

The issues were broadly with the Meeting Calendar plugin, Sensei, and hardcoded styles on blocks.

- The meeting calendar could probably be fixed by updating the wp-scripts package there, which would create RTL styles (but the 15 -> 27 version jump would probably cause other issues).
- Sensei's RTL support is limited, see https://github.com/Automattic/sensei/issues/1836, so there are overrides in the `_rtl.scss` file.
- The Learning Pathway Card sets the padding on the group block directly, and since it's loaded via a template, the `convert_inline_style_to_rtl` never runs to flip those values (hopefully the `the_block_template_html` filter will be added in 6.7, which will be a better place to run that filter) — in any case, I've added some CSS to `_rtl.scss` for that case as well.

**Screenshots**

| | Before | After |
|---|---|---|
| [Lesson page](https://learn.wordpress.org/lesson/how-to-find-help-with-wordpress-2/) |  | Screen Shot 2024-07-11 at 16 50 11 |
| [Learning Pathway homepage](https://learn.wordpress.org/learning-pathways/) |  | Screen Shot 2024-07-11 at 16 57 09 |
| [Online workshops](https://learn.wordpress.org/online-workshops/) |  | Screen Shot 2024-07-11 at 16 58 56 |
| [Course homepage](https://learn.wordpress.org/course/beginner-wordpress-developer/) | Screen Shot 2024-07-11 at 17 03 53 | Screen Shot 2024-07-11 at 16 59 43 |
| [Quiz](https://learn.wordpress.org/quiz/wordpress-taxonomies-2/) |  | Screen Shot 2024-07-11 at 17 03 01 |

### CODE REVIEW

File: wp-content/themes/pub/wporg-learn-2024/src/style/_rtl.scss [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+	.sensei-lms-quiz-back-to-lesson svg {
+		transform: rotate(180deg);
+	}
+
+	// Override the space between quiz inputs and question labels.
+	&.quiz form #sensei-quiz-list li ul li input {
+		margin: 0 10px 0 0;
+	}
+
+	// Fix the border on the Sensei lesson page.
```
Comment (adamwoodnz) [inline wp-content/themes/pub/wporg-learn-2024/src/style/_rtl.scss]: This border isn't actually supposed to be displayed; it's a Sensei thing and the design doesn't show it. An [issue](https://github.com/WordPress/Learn/issues/2601) was opened and closed to be addressed in https://github.com/WordPress/Learn/issues/2638. So I think we can remove this and unset the border and padding for all modes.

**Comment by adamwoodnz** [comment]
The My Courses filter still need to be switched to the opposite side

Screenshot 2024-07-12 at 11 38 54 AM

**Comment by adamwoodnz** [comment]
Quiz radio inputs look good but checkboxes are misaligned, see https://learn.wordpress.org/quiz/how-to-use-headings-for-accessibility-and-seo/

Screenshot 2024-07-12 at 11 43 57 AM

**Comment by adamwoodnz** [comment]
The images in the Learning Pathway cards get cropped from the wrong side on smaller screens (screenshots are at iPad landscape size)

| LTR | RTL |
|--------|--------|
| learn wordpress org_learning-pathways__locale=art_xpirate(iPad) | learn wordpress org_learning-pathways_(iPad) | 

And on the homepage some borders need switching

learn wordpress org__locale=ar(iPad)

Screenshot 2024-07-12 at 11 54 39 AM

**Review by adamwoodnz — COMMENTED** [review]
Looking much better, but I found a few more things sorry

File: wp-content/themes/pub/wporg-learn-2024/src/style/_rtl.scss (hunk 1 above)
Comment (ryelle) [reply wp-content/themes/pub/wporg-learn-2024/src/style/_rtl.scss]: Removed it in https://github.com/WordPress/Learn/pull/2669/commits/62adb0cf2817e868147bd0be0e9feb24975478dc

| LTR | RTL |
|---|---|
| Screenshot 2024-07-15 at 1 39 27 PM | Screenshot 2024-07-15 at 1 39 16 PM |

**Comment by ryelle** [comment]
> The My Courses filter still need to be switched to the opposite side

This looks correct for me:
Screenshot 2024-07-15 at 1 09 31 PM

> And on the homepage some borders need switching

I can't replicate this, borders look fine to me: 
Screenshot 2024-07-15 at 1 27 15 PM

Do you have an out of date wporg-parent on your sandbox, maybe? These are both things that [the RTL filters](https://github.com/WordPress/wporg-parent-2021/blob/trunk/source/wp-content/themes/wporg-parent-2021/inc/gutenberg-tweaks.php) should fix.

Quizzes with checkboxes have been fixed (https://github.com/WordPress/Learn/pull/2669/commits/6d12c9efba2f29ccf43586efe19ec068602b0041)

Screen Shot 2024-07-15 at 13 40 52

And I've flipped the image alignment on learning pathways cards for RTL sites (https://github.com/WordPress/Learn/pull/2669/commits/c53f67c0926c21791d6b7f6c8778b6426fd3f33d)

Screen Shot 2024-07-15 at 13 41 20

**Review by adamwoodnz — APPROVED** [review]
LGTM now except for the border styles. I've updated all on my sandbox and still seeing them. The culprit seems to be https://learn.wordpress.org/wp-content/plugins/gutenberg/build/block-library/common-rtl.css?ver=18.8.0

Screenshot 2024-07-18 at 11 46 24 AM

If you're not seeing it then maybe ignore.

**Comment by ryelle** [comment]
Aha, I managed to trigger it by preventing inline styles — I was seeing this as an inline style, but it's using the unflipped LTR version here (that seems like a different problem 😬 )

Screenshot 2024-07-18 at 6 03 58 PM

This style should not be flipped in the RTL stylesheet, and is [a bug in gutenberg](https://github.com/WordPress/gutenberg/pull/44170)- by setting the `border-right-style` it inherits default a width & color; and since the left style is not set, it's not visible.

This can be worked around by setting the right style (to make the inline-end border visible) and the left width 0 (to hide the inline-start border). Left/right are flipped in the RTL function in the parent theme, so it's easier to talk in logical property direction 🙂

**Pull request merged**